docs/DATA_MODEL.md

```diff
@@ -0,0 +1,299 @@
+while the VariantName "foo" is the same for both Variants, the values "bar" and "bizz" are different and "bar" has to
+come before "bizz".
+
+Empty VariantValues, like `lang=` mean that a Variant is NOT available, and is the same as it being not here. So in the
+canonical form empty VariantValues are removed so that for example `description:lang=` will become `description`. Empty
+VariantValues are used in [KeyFilters](#keyfilters).
+
+Empty VariantNames on the other hand are invalid and will be rejected as an error. TODO do we?
+
+If you are using Boudicca's libraries to work with Variants the library will make sure all keys are canonical, and our
+EventDB will also make sure that all ingested keys are changed to be canonical. That also means that all keys you get
+from our EventDB and our Search Service are guaranteed to be canonical.
+
+## KeyFilters
+
+When you are working with Properties which have multiple Variants or are using the [Boudicca Query Language](QUERY.md)
+you often want to select some subset of all available PropertyVariants of one Property for querying or displaying.
+To help you with this Boudicca offers support for so-called KeyFilters, which look like normal Keys.
+One example would be `description:lang=de`, which is a filter filtering for the PropertyName `description` and the
+Variant `lang=de`.
+This is done by taking all available PropertyVariants for the Property `name` and then only taking those whose `lang`
+Variant matches the value `de`. It does not matter if the PropertyVariant has additional Variants, those will be
+ignored.
+
+If KeyFilters return more than one PropertyVariant, the result will be sorted alphabetically by
+their [Canonical Form](#canonical-form).
+
+One difference compared to Keys is the empty VariantValue, like `lang=`. This means "select all PropertyVariants which
+do NOT have the VariantName at all".
+Another difference is that `*` is allowed as a PropertyName, this would match and select all Properties. You can still
+add Variant Filters to the `*` filter.
+
+Let's look at some examples given following PropertyVariants of the `description` Property:
+
+* `description`
+* `description:lang=de`
+* `description:lang=de:format=markdown`
+* `description:lang=en`
+* `description:lang=en:format=markdown`
+
+With the filter `description:lang=de` you would get the two PropertyVariants `description:lang=de`
+and `description:lang=de:format=markdown`.
+
+With the filter `description` you would get all PropertyVariants as a result, because there is no filter except the
+PropertyName.
+
+With the empty VariantValue filter `description:lang=` you will only get `description` as a result, since this is the
+only PropertyVariant without the `lang` Variant.
+
+## Unknown Variants
+
+Boudicca will ignore all Variants it does not understand and treat the keys as if those Variants are not here. This
```

Comment:
[nit[ as a noob, i might understand why it does not work - http statuscode and/or message on requests level would be beneficial to avoid wasting some more time finding out what the issue is

Reply:
Hm, just thinking out loud - either ignore it - or return no results at all. I totally think this/current approach is totally fine

Reply:
well, the reason i want to ignore and not reject unknown variants is that maybe a user wants to add some new variant to his data. it also makes it easier to introduce new known variants because old clients will still work